packages/kbn-tinymath/grammar/grammar.peggy

```diff
@@ -70,45 +96,55 @@ Variable
 
 // expressions
 
+// An Expression can be of 3 different types:
+// * a Comparison operation, which can contain recursive MathOperations inside
```

Comment:
It's pretty smart to do this within the parsing step, but I worry it's too clever (hard to maintain later on) and it also doesn't create great error messages:
<img width="773" alt="Screenshot 2022-10-04 at 17 58 09" src="https://user-images.githubusercontent.com/1508364/193868068-cfabe48c-daff-4784-a4e9-5beea9a3598c.png">

Maybe it makes more sense to pull the type check logic into a separate step _after_ the parsing? Walking the tree and keeping track of the type while recursing should work fine.

Reply:
The error is generated at validation time while traversing the AST.
The grammar works fine for the expression `ifelse(unique_count(extension, kql='...') == 1, count() > 2)` => try to paste the `peggy` file content in here to see it passing: https://peggyjs.org/online.html

Reply:
Latest commit fixed the validation bug:

<img width="351" alt="Screenshot 2022-10-05 at 15 22 41" src="https://user-images.githubusercontent.com/924948/194071651-fc497a42-8180-4dc3-98ae-be4c8e41a25c.png">